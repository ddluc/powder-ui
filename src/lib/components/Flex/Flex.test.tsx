import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Flex, FlexProps } from './index';

// Define unit tests
describe('Flex', () => {
  afterEach(cleanup);

  describe('as parent', () => {
    const args: FlexProps = {
      wrap: 'wrap',
      flow: 'row wrap',
      justifyContent: 'flex-start',
      alignItems: 'flex-end',
      alignContent: 'space-between',
      gap: '10px'
    };
    it('should render with defaults', () => {
      const result = render(<Flex />);
      const component = result.container.firstChild;
      expect(component).toMatchSnapshot();
    });
    it('should render with defined props for flex parent', () => {
      const result = render(<Flex row {...args} />);
      const component = result.container.firstChild;
      expect(component).toMatchSnapshot();
    });
    it('should render with flex-direction row', () => {
      const result = render(<Flex row {...args} />);
      const component = result.container.firstChild;
      expect(component).toHaveStyleRule('flex-direction', 'row');
    });

    it('should render with flex-direction column', () => {
      const result = render(<Flex column {...args} />);
      const component = result.container.firstChild;
      expect(component).toHaveStyleRule('flex-direction', 'column');
    });

    it('should render with flex-direction row-reverse', () => {
      const result = render(<Flex row reverse {...args} />);
      const component = result.container.firstChild;
      expect(component).toHaveStyleRule('flex-direction', 'row-reverse');
    });

    it('should render with flex-direction column-reverse', () => {
      const result = render(<Flex column reverse {...args} />);
      const component = result.container.firstChild;
      expect(component).toHaveStyleRule('flex-direction', 'column-reverse');
    });
  });

  describe('as child', () => {
    const args: FlexProps = {
      flex: '1 2 3',
      grow: 1,
      shrink: 2,
      basis: 3,
      order: 5,
      alignSelf: 'baseline'
    };
    it('should render with defined props', () => {
      const result = render(<Flex row {...args} />);
      const component = result.container.firstChild;
      expect(component).toMatchSnapshot();
    });
    it('should "center"', () => {
      const result = render(<Flex row center />);
      const component = result.container.firstChild;
      expect(component).toHaveStyleRule('justify-content', 'center');
      expect(component).toHaveStyleRule('align-items', 'center');
    });
  });
});
