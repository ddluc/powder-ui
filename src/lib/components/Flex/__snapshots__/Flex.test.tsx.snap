// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex as child should render with defined props 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1 2 3;
  -ms-flex: 1 2 3;
  flex: 1 2 3;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 2;
  -ms-flex-negative: 2;
  flex-shrink: 2;
  -webkit-flex-basis: 3;
  -ms-flex-preferred-size: 3;
  flex-basis: 3;
  -webkit-order: 5;
  -ms-flex-order: 5;
  order: 5;
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
}

<div
  class="c0 c1"
  order="5"
/>
`;

exports[`Flex as parent should render with defaults 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  class="c0 c1"
/>
`;

exports[`Flex as parent should render with defined props for flex parent 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  gap: 10px;
}

<div
  class="c0 c1"
  wrap="wrap"
/>
`;
